src/checker/args.rs

```diff
@@ -0,0 +1,237 @@
+        let config = load_config(popts.opt_str("config-file"), popts.opt_str("checks"))?;
+
+        if popts.opt_present("help-config") {
+            print_config_supplied(config);
+            std::process::exit(0);
+        }
+
+        Ok(Args {
+            mz_url: popts.opt_get_default(
+                "materialized-url",
+                "postgres://ignoreuser@materialized:6875/materialize".to_owned(),
+            )?,
+            only_initialize: popts.opt_present("only-initialize"),
+            config,
+        })
+    }
+}
+
+fn load_config(config_path: Option<String>, cli_checks: Option<String>) -> Result<Config, Error> {
+    // load and parse th toml
+    let config_file = config_path
+        .as_ref()
+        .map(std::fs::read_to_string)
+        .unwrap_or_else(|| Ok(DEFAULT_CONFIG.to_string()));
+    let mut config = match &config_file {
+        Ok(contents) => toml::from_str::<Config>(&contents).map_err(|e| {
+            format!(
+                "Unable to parse config file {}: {}",
+                config_path.as_deref().unwrap_or("DEFAULT"),
+                e
+            )
+        })?,
+        Err(e) => {
+            eprintln!(
+                "unable to read config file {:?}: {}",
+                config_path.as_deref().unwrap_or("DEFAULT"),
+                e
+            );
+            std::process::exit(1);
+        }
+    };
+
+    // filter to only things enabled in the command line OR the config file
+    if let Some(checks) = cli_checks {
+        let enabled_cs: Vec<_> = checks.split(',').collect();
+        if !enabled_cs.is_empty() {
+            config
+                .checks
+                .retain(|c| enabled_cs.contains(&c.name.as_str()));
+        }
+    } else {
+        config.checks.retain(|c| c.enabled);
+    }
+
+    // Sort checks by specificity (number of columns involved in the check)
+    for c in &mut config.checks {
+        c.rows.sort_by(|a, b| {
+            let a_length = a.columns.len();
+            let b_length = b.columns.len();
+            if a_length < b_length {
+                Ordering::Greater
+            } else if a_length == b_length {
+                Ordering::Equal
+            } else {
+                Ordering::Less
+            }
+        });
+    }
+
+    Ok(config)
+}
+
+fn print_config_supplied(config: Config) {
+    println!("named checks:");
+    for c in config.checks {
+        println!(
+            "    thread_count={} sleep={:?} query={}",
+            c.thread_count.unwrap_or(config.default_check.thread_count),
+            c.sleep_ms.unwrap_or(config.default_check.sleep_ms),
+            c.name
+        );
+    }
+}
+
+// inner parsing helpers
+
+/// Config File
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    /// Default to be filled in for other queries
+    pub default_check: DefaultCheck,
+    /// Checks are instantiated as views which are polled continuously
+    pub checks: Vec<Check>,
+    // Sources
+    pub sources: Vec<Source>,
```

Comment:
Do you envision people running this against multiple kafka brokers/schema registries? I know we don't for the first toml you've introduced. If not, you could simplify the code around sources by switching this to just:

```
pub sources: Source,
```